Library/Homebrew/test/support/helper/spec/integration_command.rb

```diff
@@ -0,0 +1,92 @@
+require "open3"
+
+RSpec::Matchers.define_negated_matcher :not_to_output, :output
+RSpec::Matchers.define_negated_matcher :be_a_failure, :be_a_success
+
+RSpec::Matchers.define :be_a_success do
+  match do |actual|
+    if actual.is_a?(Proc)
+      status = actual.call
+    else
+      status = actual
+    end
+
+    status.success?
+  end
+
+  def supports_block_expectations?
+    true
+  end
+
+  # It needs to be nested like this:
+  #
+  #   expect {
+  #     expect {
+  #       # command
+  #     }.to be_a_success
+  #   }.to output(something).to_stdout
+  #
+  # rather than this:
+  #
+  #   expect {
+  #     expect {
+  #       # command
+  #     }.to output(something).to_stdout
+  #   }.to be_a_success
+  #
+  def expects_call_stack_jump?
+    true
+  end
+end
+
+RSpec.configure do |config|
+  config.before(:each) do |example|
+    (HOMEBREW_PREFIX/"bin").mkpath
+    FileUtils.touch HOMEBREW_PREFIX/"bin/brew"
+    @example = example
+  end
+
+  config.after(:each) do
+    FileUtils.rm HOMEBREW_PREFIX/"bin/brew"
+    FileUtils.rmdir HOMEBREW_PREFIX/"bin"
+  end
+
+  config.include(
+    Module.new do
+      def command_id_from_args(args)
+        @cmd_id_index ||= 0
+        pretty_args = args.join(" ").gsub(TEST_TMPDIR, "@TMPDIR@")
+        description = @example.full_description
+        "[#{description}][#{@cmd_id_index += 1}] brew #{pretty_args}"
+      end
+
+      def brew(*args)
```

Comment:
I think this method is sufficiently complex that it could use a comment explaining what it does.